src/dawn/IIR/DoMethod.h

```diff
@@ -94,6 +94,8 @@ class DoMethod : public IIRNode<Stage, DoMethod, StatementAccessesPair> {
   /// mergeWithDoInterval is true
   boost::optional<Interval> computeEnclosingAccessInterval(const int accessID,
                                                            const bool mergeWithDoInterval) const;
+
+  inline virtual void updateFromChildren() override {}
```

Comment:
The main API for the user is the update() function of the IIRNode, that will end up calling this (if required). 
All nodes of the tree are forced to implement an `updateLevel` and `updateFromChildren` methods that will compute the DerivedInfo from your own IIR level data or from the children data. 
The typical example is the Field of the MultiStage that is computed in updateFromChildren from the Fields of all children Stages. 
The DoMethod does not construct a derived info from the children (at the moment) so that's why it is empty. It is a virtual method, we could omit it, but I left it there to make it more explicit. 

I added doc to these functions in the PR VI